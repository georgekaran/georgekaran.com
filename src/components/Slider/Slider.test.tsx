import '@/test/mock-match-media'
import React from 'react'

import { Slider } from '.'
import { renderWithTheme } from '@/test/helpers'

describe('<Slider />', () => {
  it('should render with 1 active item', () => {
    const { container } = renderWithTheme(
      <Slider>
        <h1 key="1">1</h1>
        <h1 key="2">2</h1>
        <h1 key="3">3</h1>
      </Slider>
    )
    expect(container.querySelectorAll('.slick-slide')).toHaveLength(7)
    expect(container.querySelectorAll('li.slick-active')).toHaveLength(1)
  })
})
