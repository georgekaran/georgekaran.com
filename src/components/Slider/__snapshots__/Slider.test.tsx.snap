// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Slider /> should match snapshot 1'] = `
.c0 .slick-slider {
  position: relative;
  display: block;
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-touch-callout: none;
  -khtml-user-select: none;
  -ms-touch-action: pan-y;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
}

.c0 .slick-list {
  position: relative;
  display: block;
  overflow: hidden;
  margin: 0;
  padding: 0;
}

.c0 .slick-list:focus {
  outline: none;
}

.c0 .slick-list.dragging {
  cursor: pointer;
  cursor: hand;
}

.c0 .slick-slider .slick-track,
.c0 .slick-slider .slick-list {
  -webkit-transform: translate3d(0,0,0);
  -moz-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  -o-transform: translate3d(0,0,0);
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c0 .slick-track {
  position: relative;
  top: 0;
  left: 0;
  display: block;
}

.c0 .slick-track:before,
.c0 .slick-track:after {
  display: table;
  content: '';
}

.c0 .slick-track:after {
  clear: both;
}

.c0 .slick-loading .slick-track {
  visibility: hidden;
}

.c0 .slick-slide {
  display: none;
  float: left;
  height: 100%;
  min-height: 1px;
}

.c0 .slick-slide > div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 [dir='rtl'] .slick-slide {
  float: right;
}

.c0 .slick-slide img {
  display: block;
}

.c0 .slick-slide.slick-loading img {
  display: none;
}

.c0 .slick-slide.dragging img {
  pointer-events: none;
}

.c0 .slick-initialized .slick-slide {
  display: block;
}

.c0 .slick-loading .slick-slide {
  visibility: hidden;
}

.c0 .slick-vertical .slick-slide {
  display: block;
  height: auto;
  border: 1px solid transparent;
}

.c0 .slick-arrow.slick-hidden {
  display: none;
}

.c0 .slick-dots {
  list-style: none;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 2.4rem;
}

.c0 .slick-dots li {
  background: #737380;
  width: 0.8rem;
  height: 0.8rem;
  border-radius: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 0.8rem;
  cursor: pointer;
}

.c0 .slick-dots li.slick-active {
  background: #28F5BE;
}

.c0 .slick-dots button {
  opacity: 0;
  width: 0.8rem;
  height: 0.8rem;
  cursor: pointer;
}

<section
  class="c0"
>
  <div
    class="slick-slider slick-initialized"
    dir="ltr"
  >
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1;"
      >
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px; position: relative; left: 0px; opacity: 1; transition: opacity 1000ms ease, visibility 1000ms ease;"
          tabindex="-1"
        >
          <div>
            <h1
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              1
            </h1>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide"
          data-index="1"
          style="outline: none; width: 0px; position: relative; left: 0px; opacity: 0; transition: opacity 1000ms ease, visibility 1000ms ease;"
          tabindex="-1"
        >
          <div>
            <h1
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              2
            </h1>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide"
          data-index="2"
          style="outline: none; width: 0px; position: relative; left: 0px; opacity: 0; transition: opacity 1000ms ease, visibility 1000ms ease;"
          tabindex="-1"
        >
          <div>
            <h1
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              3
            </h1>
          </div>
        </div>
      </div>
    </div>
    <ul
      class="slick-dots"
      style="display: block;"
    >
      <li
        class="slick-active"
      >
        <button>
          1
        </button>
      </li>
      <li
        class=""
      >
        <button>
          2
        </button>
      </li>
      <li
        class=""
      >
        <button>
          3
        </button>
      </li>
    </ul>
  </div>
</section>
`
